modules/thermal_hydraulics/src/components/HeatTransferFromHeatStructure3D1Phase.C

```diff
@@ -43,19 +43,15 @@ HeatTransferFromHeatStructure3D1Phase::HeatTransferFromHeatStructure3D1Phase(
     _flow_channel_names(getParam<std::vector<std::string>>("flow_channels")),
     _boundary(getParam<BoundaryName>("boundary")),
     _hs_name(getParam<std::string>("hs")),
-    _fch_alignments(_flow_channel_names.size(), nullptr),
+    _fch_alignment(_mesh),
     _layered_average_uo_direction(MooseEnum("x y z"))
 {
   for (const auto & fch_name : _flow_channel_names)
     addDependency(fch_name);
   addDependency(_hs_name);
 }
 
-HeatTransferFromHeatStructure3D1Phase::~HeatTransferFromHeatStructure3D1Phase()
-{
-  for (auto & fcha : _fch_alignments)
-    delete fcha;
-}
+HeatTransferFromHeatStructure3D1Phase::~HeatTransferFromHeatStructure3D1Phase() {}
```

Comment:
If the d-tor does nothing, then just delete it. The compiler will build the default one for us.